source/application/BioXAS/BioXASAppController.cpp

```diff
@@ -635,6 +631,16 @@ AMScanConfigurationViewHolder3* BioXASAppController::createScanConfigurationView
 	return configurationViewHolder;
 }
 
+void BioXASAppController::addComponentView(QObject *component, const QString &viewName, const QString &categoryName, const QString &icon)
```

Comment:
It's possible, we can take a couple of moments to investigate together if you like.